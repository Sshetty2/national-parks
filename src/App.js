import React, { Component } from 'react';
import './App.css';
import ParkMap from './ParkMap.js';
import FilterControls from './FilterControls.js';
import LandingPage from './LandingPage.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      parks: [],
      usStates: {},
      currentParksToShow: [],
      vistedParks: [],
      bucketListParks: [],
      pageStatus: 'landing',
      randomImageClass: `landing-background${Math.floor(Math.random() * 6)}`
    };
  }

  showAllParks = (event) => {
    event.preventDefault();
    this.setState({
      currentParksToShow: this.state.parks
    })
  }

  showVisitedParks = (event) => {
    event.preventDefault();
    let visitedParks;

    if (localStorage.hasOwnProperty('visitedParks')) {
      let cachedVistedParkKeys = localStorage.getItem('visitedParks');
      let vistedParksKeys = JSON.parse(cachedVistedParkKeys);
      visitedParks = this.state.parks.filter(park => {
        return vistedParksKeys.includes(park.urlCode)
      });

      this.setState({
        currentParksToShow: visitedParks
      });
    }
  }

  showBucketList = (event) => {
    event.preventDefault();
    let bucketListParks;

    if (localStorage.hasOwnProperty('bucketList')) {
      let cachedBucketListKeys = localStorage.getItem('bucketList');
      let bucketListKeys = JSON.parse(cachedBucketListKeys);
      
      bucketListParks = this.state.parks.filter( park => {
        return bucketListKeys.includes(park.urlCode)
      });
  
      this.setState({
        currentParksToShow: bucketListParks
      });
    } 
  }

  componentDidMount() {
    fetch("https://whateverly-datasets.herokuapp.com/api/v1/nationalParks1810")
      .then(data => data.json())
      .then(results => {
        results.nationalParks1810.forEach(park => {
          if (park.parkName === 'Sequoia') {
            park.urlCode = 'seki2';
          }
        });
        this.setState({
          parks: results.nationalParks1810,
          currentParksToShow: results.nationalParks1810
        });
      })
      .catch(error => console.log(error));

    fetch("https://whateverly-datasets.herokuapp.com/api/v1/states1810")
      .then(data => data.json())
      .then(results => {
        this.setState({
          usStates: results.states1810
        });
      })
      .catch(error => console.log(error));
  }

  openHomePage = () => {
    this.setState({
      pageStatus: 'home'
    });
  }

  render() {
    switch(this.state.pageStatus) {
      case('home'):
        return (
          <div className={this.state.randomImageClass}>
            <div className="overlay">
              <h1 className="home-title">Mark My Parks</h1>
              <ParkMap parks={this.state.currentParksToShow}/>
              <div className="filters">
                <button onClick={this.showAllParks}>Show All Parks</button>
                <button onClick={this.showVisitedParks}>Show Visited Parks</button>
                <button onClick={this.showBucketList}>Show Bucket List Parks</button>
              </div>
              <FilterControls usStates={this.state.usStates} />
            </div>
          </div>
        );
      default:
        return (
          <div className={this.state.randomImageClass}>
            <LandingPage openHomePage={this.openHomePage} />
          </div>
        );
    }
  }
}

export default App;
