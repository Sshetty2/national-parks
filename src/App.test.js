import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the proper default state', () => {
    expect(wrapper.state()).toEqual({
      parks: [],
      usStates: {},
      vistedParks: [],
      currentParksToShow: [],
      bucketListParks: [],
      pageStatus: 'landing',
      // need to check for randomImageClass - not sure how to test a random number in state.
    })
  });
  
});