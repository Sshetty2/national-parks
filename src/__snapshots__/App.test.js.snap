// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="landing-background3"
>
  <LandingPage
    openHomePage={[Function]}
  />
</div>
`;
